### ISSUE TITLE
Inconsistent results when multiprocessing is used to aligning words

### ISSUE THREAD

**Issue by izaszopa** [MontrealCorpusTools/Montreal-Forced-Aligner#501]
Hello, I decided to write here this question because I cannot find answer as to why this situation takes place in the calculations.
I'm using MFA validate and align with english_mfa dictionary. My concern is connected to results that I get while generating results for one single file at a time and when this file is calculated with many audio files with few speakers in input folder. The difference is not very big, but I don't think there should be any so I don't know why is it there. Can someone explain it to me?
Example results and their differences:
![difference_multiprocessing](https://user-images.githubusercontent.com/42964730/185144036-3205156c-8f2b-4fec-9e30-1b1357c57fa4.png)
I marked with yellow different results, on left files were run one by one, and on right all together.

**Comment by rohun-tripathi** [comment]
I see an issue that might be the cause of the issue above. I am not sure why my issue occurs.
My issue: I am observing that the alignment seems to improve as I increase the size of the data set I annotate.

I suspect that it has some thing to do with speaker adaption. Does the performance improve if more samples of a speaker are used?

**Comment by mmcauliffe** [comment]
Yes, exactly, there are two passes that MFA does.  The first uses a speaker-independent model to generate alignments, and then the second pass uses those alignments to estimate per-speaker transforms and re-align with the speaker adapted model.  This second pass can move the boundaries around a little bit, and align more files in general, because the initial alignments can bootstrap better speaker-adapted features for the second pass.

**Comment by rohun-tripathi** [comment]
Great, thanks for the quick response!

Follow-up question - If I have a dataset of 5000 clips, with five speakers of 1000 clips each.
Is there a chance that running it on all the clips as one speaker might do better than annotating the speakers and running the MFA with speaker labels?

I know this question is dataset dependent. I am curious based on your experience

**Comment by mmcauliffe** [comment]
I would keep them as distinct speakers, with 1000 files, it should be a decent amount of data per speaker for estimation.  Even if they're the same gender, their feature spaces are going to be naturally different, so lumping them together *might* be basically the same as not having speaker information at all (haven't tested it and certainly depends on the data).
